path_role.go

```diff
@@ -334,11 +335,18 @@ func (b *kubeAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logical
 		return logical.ErrorResponse("can not mix %q with values", "*"), nil
 	}
 
-	// optional audience field
+	// audiences will be required in kubernetes roles in a future Vault version
 	if audience, ok := data.GetOk("audience"); ok {
 		role.Audience = audience.(string)
 	}
 
+	// Vault 1.21+ will require an audience to be set on a role for security reasons.
+	// Log a warning if the role does not specify an audience.
+	if strings.TrimSpace(role.Audience) == "" {
+		b.Logger().Warn("Role '%s' does not specify an audience. In Vault v1.21+, specifying an audience will be required.", roleName)
```

Comment:
This is incorrect usage of the Warn method. Everything after the first argument must be key/value pairs.

```
2025-06-16T10:30:21.943-0500 [WARN]  Role '%s' does not specify an audience. In Vault v1.21+, specifying an audience will be required.: EXTRA_VALUE_AT_END=plugin-test
```